### PR TITLE
feat: bump `@metamask/keyring-controller` to v6.0.0

### PR DESCRIPTION
**Pull request by gantunesr**
## Description

❗ This is a new PR, the original one is https://github.com/MetaMask/metamask-mobile/pull/6770 ❗ 

Bump `@metamask/keyring-controller` from `v1.0.1` to `v6.0.0`.

- Patches
  - `micro-ftch+0.3.1`: Added to comment out a specific node logic that was causing a compatibility issue with another library.
  - `@metamask+message-manager+7.0.1`: ~Updated since we're using the same version for both `SignatureController` and `KeyringController`.~ Removed. Reference commit [e5b9010](https://github.com/MetaMask/metamask-mobile/pull/7267/commits/e5b9010173d44a3d44aada8928cd80bf27b9f55a).
  - `@metamask+message-manager+1.0.1`: Removed. Reference commit [e5b9010](https://github.com/MetaMask/metamask-mobile/pull/7267/commits/e5b9010173d44a3d44aada8928cd80bf27b9f55a).
  - `@metamask+signature-controller++@metamask+message-manager+7.0.1.patch`: Removed. Reference commit [e5b9010](https://github.com/MetaMask/metamask-mobile/pull/7267/commits/e5b9010173d44a3d44aada8928cd80bf27b9f55a).
- UI/UX changes
  - To reveal the SRP or Private Key, the CTA to verify the password will be disabled until a string of 8 characters or more is provided. After it, the CTA will be enabled and will call to verify the password against the KeyringController and provide a result.

## Test cases

- [ ] Onboarding
  - [ ] Create new wallet
  - [ ] Import SRP
- [ ] Authentication - Lock and unlock wallet using different authentication mechanisms (password, fingerprint, face ID, device pin)
- [ ] Reveal SRP
- [ ] Reveal private key
- [ ] Import private key - Creation of Simple Keyring
  - [ ] Upgrade the app from previous version and import private key
- [ ] Import QR wallet accounts - Creation of QR Keyring
  - [ ] Upgrade the app from previous version and import QR wallet accounts 
- [ ] Send a simple transaction using a HD account and QR account
- [ ] Signature methods (Including QR wallet)
  - [ ] Eth Sign
  - [ ] Personal Sign
  - [ ] Sign Typed Data
  - [ ] Sign Typed Data V3
  - [ ] Sign Typed Data V4 
- [ ] Delete imported account (Simple Keyring)
- [ ] Disconnect and re-connect Keystone (QR Keyring) 
- [ ] Cancel transaction 
- [ ] Speed up the transaction
- [ ] Rename accounts
- [ ] Import private key and upgrade the app 

## Screenshots/Recordings

https://github.com/MetaMask/metamask-mobile/assets/17601467/3050db0a-8951-4c27-aaf1-85ff6d281ef5

## Issue

Fixes https://github.com/MetaMask/mobile-planning/issues/755

## Checklist

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Comment by gantunesr** [comment]
@SocketSecurity ignore prismjs@1.28.0

**Comment by olenapankina** [comment]
**Tested on**
iPhone Xr iOS 16.1.1
Samsung Galaxy M21 Android 12, One UI Core version 4.1
commit 685c87f23011f219cf579b6a383e44a3f50d65c1

**Issue_01:** HD account is not saved
**Steps to reproduce:**

1. Import an account using SRP
2. Add new account
3. Close the app - reopen
4. Observe only one account present

**Actual result:** Only imported account is present
**Expected result:** both account are present

Note: after tapping on add new account again - account with the same address is added

https://github.com/MetaMask/metamask-mobile/assets/116747263/b96a0819-39c7-4418-a6b7-7556a955b9ed

**Comment by gantunesr** [comment]
Hi @olenapankina, is there a step I'm missing here? I can't reproduce the bug


https://github.com/MetaMask/metamask-mobile/assets/17601467/b385b5f7-2ddf-441e-977f-e98a5b9a42f7

**Comment by plasmacorral** [comment]
> Hi @olenapankina, is there a step I'm missing here? I can't reproduce the bug
> 
>  Screen.Recording.2023-09-23.at.22.43.10.mov

Could it be related to the build somehow?  I notice that Olena was testing with a QA build, and looks like you were running a local build.

**Comment by stanleyyconsensys** [comment]
i have the similar exp, that my imported token has gone alway
i guess this is something to do with the keyring controller init state
in our ledger repo, we init the keyring state with below way

```
const additionalKeyrings = [QRHardwareKeyring.type, LedgerKeyring.type];
const keyringState = {
      keyringTypes: additionalKeyrings,
      ...initialState.KeyringController,
      ...initialKeyringState,
    };
```

**Comment by stanleyyconsensys** [comment]
@gantunesr i think you have to test it with import account

i also exp the similar issue, and i debug on the raw code, and i found that
-  the type [simple key ring] doesnt included into the keyringBuilders from eth keyring controller, instead it become undefined type
- hence when calling get accounts from eth keyring controller, it cant import/fetch the account type that match the import address 

im not sure is it sth related to the init state of keyring controller from engine.tx

note that: import address will using [simple key ring] type

**Comment by olenapankina** [comment]
@gantunesr here's a recording from local build alongside metro logs

https://github.com/MetaMask/metamask-mobile/assets/116747263/5107542d-af5f-42f0-840a-f6fcb70ce825

**Comment by gantunesr** [comment]
@olenapankina the bug should be fixed

**Comment by olenapankina** [comment]
**Tested on** 
iPhone Xr iOS 16.1.1
Samsung Galaxy M21 Android 12, One UI Core version 4.1
bitrise build 9971

Issue_01 confirmed fixed. Test cases listed in this ticket description passed.
**QA Passed**

**Comment by gantunesr** [comment]
@SocketSecurity ignore debug@4.1.1

**Review by mikesposito — APPROVED** [review]
LGTM!

**Review by montelaidev — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -565,6 +566,8 @@
       "wdio-image-comparison-service>webdriver-image-comparison>canvas": false,
       "appium-adb>@appium/support>sharp": true,
       "react-native-svg-asset-plugin>sharp": true,
+      "@metamask/keyring-controller>@keystonehq/metamask-airgapped-keyring>@keystonehq/base-eth-keyring>@ethereumjs/tx>ethereumjs-util>ethereum-cryptography>keccak": false,
+      "@metamask/keyring-controller>@keystonehq/metamask-airgapped-keyring>@keystonehq/base-eth-keyring>@ethereumjs/tx>ethereumjs-util>ethereum-cryptography>secp256k1": false,
```
[the remark this answers is not in the document]
Comment (leotm) [reply package.json:570]: for context i'm reviewing:
- https://github.com/MetaMask/metamask-mobile/pull/7129

noting more `keccak` and `secp256k1` now being set to `false`

[the remark this answers is not in the document]
Comment (leotm) [reply package.json:570]: thx @legobeat for fixing/resolving! in
- https://github.com/MetaMask/metamask-mobile/pull/7129

**Pull request merged**